synapseclient/core/upload/upload_functions_async.py

```diff
@@ -306,6 +306,11 @@ async def upload_synapse_s3(
     return await get_file_handle(file_handle_id=file_handle_id, synapse_client=syn)
 
 
+def _get_aws_credentials() -> None:
```

Comment:
I know it's used only in testing purposes, but what was happening before?

Reply:
You could only specify the aws account to use by profile name. Then that means you have to have the correct .aws config available on disk.

Doing it this way allows me to mock which credentials are being used easier.

I don't think there was a test case testing this before, so it was not an issue.